libraries/AC_Avoidance/AC_Avoid.cpp

```diff
@@ -1193,12 +1194,19 @@ void AC_Avoid::adjust_velocity_proximity(float kP, float accel_cmss, Vector3f &d
         
             break;
         }
-      
+
         case BEHAVIOR_STOP: {
             // vector from current position to obstacle
             Vector3f limit_direction;
-            // find intersection with line segment
-            const float limit_distance_cm = _proximity.distance_to_obstacle(i, Vector3f{}, stopping_point, limit_direction);
+            // find closest point with line segment
+            // also see if the vehicle will "roughly" intersect the boundary with the projected stopping point
+            const bool intersect = _proximity.closest_point_from_segment_to_obstacle(i, Vector3f{}, stopping_point_plus_margin, limit_direction);
+            if (!intersect) {
+                // the vehicle isn't intersecting the boundary in anyway
+                continue;
```

Comment:
Definitely not a blocker but personally I avoid using "continue".. I just find it hard to figure out where it's going to pop out.  I like simple early returns but I'd prefer we just used "if (intersect) {".